test/cadet/accounts/accounts_test.exs

```diff
@@ -204,4 +204,33 @@ defmodule Cadet.AccountsTest do
       end
     end
   end
+
+  describe "insert_or_update_user" do
+    test "existing user" do
+      user = insert(:user)
+      user_params = params_for(:user, nusnet_id: user.nusnet_id)
+      Accounts.insert_or_update_user(user_params)
+
+      updated_user =
+        User
+        |> where(nusnet_id: ^user.nusnet_id)
+        |> Repo.one()
+
```

Comment:
Prefer if you also `assert user.id == updated_user.id` to check that it's the same user.

Reply:
Done